neuro_extras/main.py

```diff
@@ -201,6 +231,155 @@ def image() -> None:
+        if url.scheme in ("s3", "gs"):
+            return UrlType.CLOUD
+        return UrlType.UNSUPPORTED
+
+
+async def _data_cp(source: str, destination: str, unpack: bool) -> None:
+    source_url = URL(source)
+    destination_url = URL(destination)
+    source_url_type = UrlType.get_type(source_url)
+    if source_url_type == UrlType.UNSUPPORTED:
+        raise ValueError(f"Unsupported source URL scheme: {source_url.scheme}")
+    destination_url_type = UrlType.get_type(destination_url)
+    if destination_url_type == UrlType.UNSUPPORTED:
+        raise ValueError(
+            f"Unsupported destination URL scheme: {destination_url.scheme}"
+        )
+
+    if source_url_type == UrlType.CLOUD and destination_url_type == UrlType.CLOUD:
+        raise ValueError(
+            "This command can't be used to copy data between cloud providers"
+        )
+    if source_url_type == UrlType.STORAGE and destination_url_type == UrlType.STORAGE:
+        raise ValueError(
+            "This command can't be used to copy data between two storage locations"
+        )
+
+    if UrlType.STORAGE in (source_url_type, destination_url_type):
+        async with neuro_api.get() as client:
+            data_copier = DataCopier(client)
+            if source_url_type == UrlType.STORAGE:
+                job = await data_copier.launch(
+                    storage_uri=source_url,
+                    src_uri=URL("/var/storage"),
+                    dst_uri=destination_url,
+                    unpack=unpack,
+                )
+            else:
+                job = await data_copier.launch(
+                    storage_uri=destination_url,
+                    src_uri=source_url,
+                    dst_uri=URL("/var/storage"),
+                    unpack=unpack,
+                )
+            while job.status == neuro_api.JobStatus.PENDING:
+                job = await client.jobs.status(job.id)
+                await asyncio.sleep(1.0)
+            async for chunk in client.jobs.monitor(job.id):
+                if not chunk:
+                    break
+                click.echo(chunk.decode(errors="ignore"), nl=False)
+            job = await client.jobs.status(job.id)
+            if job.status == neuro_api.JobStatus.FAILED:
+                logger.error("The copy job has failed due to:")
+                logger.error(f"  Reason: {job.history.reason}")
+                logger.error(f"  Description: {job.history.description}")
+                exit_code = job.history.exit_code
+                if exit_code is None:
+                    exit_code = EX_PLATFORMERROR
+                sys.exit(exit_code)
+            else:
+                logger.info("Successfully copied data")
+    else:
+        TEMP_UNPACK_DIR.mkdir(parents=True, exist_ok=True)
+        tmp_dir_name = tempfile.mkdtemp(dir=str(TEMP_UNPACK_DIR))
+        tmp_dst_url = URL.build(path=(tmp_dir_name + "/"))
+
+        await _nonstorage_cp(source_url, tmp_dst_url)
```

Comment:
IMO we need more `click.echo` on operations other than `aws cp` / `gsutil cp`, because now the logs look confusing as if we were only downloading the data into `~/.neuro/tmp`:
```
$ ne data cp gs://cookiecutter-e2e/hello.tar.gz  /tmp/asdf -x
Running gsutil -m cp -r gs://cookiecutter-e2e/hello.tar.gz /home/ay/.neuro/tmp/tmplr5hotco/
Copying gs://cookiecutter-e2e/hello.tar.gz...
/ [1/1 files][  177.0 B/  177.0 B] 100% Done                                    
Operation completed over 1 objects/177.0 B.                                      
Running tar zxvf /home/ay/.neuro/tmp/tmplr5hotco/hello.tar.gz -C /tmp/asdf
data/
data/hello.txt
Running rsync -avzh --progress /home/ay/.neuro/tmp/tmplr5hotco/ /tmp/asdf
sending incremental file list
hello.tar.gz
            177 100%    0.00kB/s    0:00:00 (xfr#1, to-chk=0/2)

sent 289 bytes  received 35 bytes  648.00 bytes/sec
total size is 177  speedup is 0.55
```

Let's cover this in a following-up PR.

Reply:
fixed as I see.